packages/grafana-ui/src/components/Table/utils.ts

```diff
@@ -208,3 +209,16 @@ export function getFilteredOptions(options: SelectableValue[], filterValues?: Se
 export function sortCaseInsensitive(a: Row<any>, b: Row<any>, id: string) {
   return String(a.values[id]).localeCompare(String(b.values[id]), undefined, { sensitivity: 'base' });
 }
+
+const replaceNonNumeric = /[^0-9.]/gi;
+export function sortNumber(rowA: Row<any>, rowB: Row<any>, id: string) {
+  const valueA = rowA.values[id];
+  const valueB = rowB.values[id];
+  const a = isInfinity(valueA) ? valueA : Number(String(valueA).replace(replaceNonNumeric, ''));
+  const b = isInfinity(valueB) ? valueB : Number(String(valueB).replace(replaceNonNumeric, ''));
+  return a === b ? 0 : a > b ? 1 : -1;
```

Comment:
this seems very expensive, for sorting 100 000 rows this is called quite a number of times. A numeric value is rendered to string, then replace non numeric, then parsed back to number. 

Is there a reason to render numeric values to strings and then replace the non numeric elements?  or is that scenario mainly for non numeric values, but cannot see any strings in your tests so not sure that can be it.

Reply:
Code was copied from `react-table` (except for the isInfinity check) and there are strings in my tests.  https://github.com/grafana/grafana/blob/d3f45719824e7c9881b5c6fe8200010cea7603ed/packages/grafana-ui/src/components/Table/utils.test.ts#L429

Question, I was under the impression that this was only called for the cells rendered so I didn't bother about performance as we don't usually render 100 000 rows. But if that is the case, then I need to optimize this.

Reply:
Ah I see, but sorting must be applied to all rows surely? Not sure how it can work by only sorting visible rows :)

Reply:
That makes sense, I'll optimize this.

Reply:
Refactored, tried to use the existing `anyToNumber` function and `toNumber` from lodash but those didn't pass the previously created tests so I created a local `toNumber` function instead.